starknet-foundry/crates/cheatable-starknet/src/rpc.rs

```diff
@@ -474,9 +482,76 @@ impl CheatableSyscallHandler<'_> {
             response_wrapper.write(vm, &mut self.syscall_handler.syscall_ptr)?;
             return Ok(());
         }
+
+        if let SyscallSelector::CallContract = selector {
+            // Increment, since the selector was peeked into before
+            self.syscall_handler.syscall_ptr += 1;
+            return self.execute_syscall(
+                vm,
+                call_contract_syscall,
+                constants::CALL_CONTRACT_GAS_COST,
+            );
+        }
+
         self.syscall_handler.execute_next_syscall(vm, hint)
     }
 
+    fn execute_syscall<Request, Response, ExecuteCallback>(
+        &mut self,
+        vm: &mut VirtualMachine,
+        execute_callback: ExecuteCallback,
+        base_gas_cost: u64,
+    ) -> HintExecutionResult
+    where
+        Request: SyscallRequest + std::fmt::Debug,
```

Comment:
Why do we need Debug trait here?

Reply:
I haven't investigated that, it is a constraint enforced by blouckifier codebase